utils/liftover.py

```diff
@@ -0,0 +1,194 @@
+    :rtype: Table
+    """
+ 
+    # filter to snps
+    ht = ht.filter(hl.is_snp(ht.alleles[0], ht.alleles[1]))
+
+    # check if reference allele matches what is in reference fasta
+    # for snps on negative strand, make sure reverse complement of ref allele matches what is in reference fasta
+    ht = ht.annotate(
+        reference_mismatch=hl.cond(
+                                ht.new_locus.is_negative_strand,
+                                (flip_base(ht.alleles[0]) != hl.get_sequence(ht.locus.contig, ht.locus.position, reference_genome=rg)),
+                                (ht.alleles[0] != hl.get_sequence(ht.locus.contig, ht.locus.position, reference_genome=rg))
+                                )
+                )
+
+    return ht
+
+ 
+def check_mismatch(ht: hl.Table) -> hl.expr.expressions.StructExpression:
+    """
+    Checks for mismatches between reference allele and allele in reference fasta
+    :param Table ht: Table to be checked
+    :return: StructExpression containing counts for mismatches and count for all variants on negative strand
+    :rtype: StructExpression
+    """
+
+    mismatch = ht.aggregate(hl.struct(total_variants=hl.agg.count(),
+                                total_mismatch=hl.agg.count_where(ht.reference_mismatch),
+                                negative_strand=hl.agg.count_where(ht.new_locus.is_negative_strand),
+                                negative_strand_mismatch=hl.agg.count_where(ht.new_locus.is_negative_strand & ht.reference_mismatch)
+                                )
+                            )
+    return mismatch
+
+
+def main(args):
+
+    hl.init()
+    
+    # get reference genomes
+    rg37 = hl.get_reference('GRCh37')
+    rg38 = hl.get_reference('GRCh38')
+
+    if args.gnomad:
+        gnomad = True
+        table_path = None
+        if not args.exomes and not args.genomes:
+            sys.exit('Error: Must specify --exomes or --genomes if lifting gnomAD ht')
+
+        if args.exomes:
+            data_type = 'exomes'
+        if args.genomes:
+            data_type = 'genomes'
+
+        logger.info('Working on gnomAD {} release ht'.format(data_type))
+        logger.info('Reading in release ht')
+        ht = get_gnomad_public_data(data_type, split=True, version=CURRENT_RELEASE)
+        logger.info('Variants in release ht: {}'.format(ht.count()))
+
+    else:
+        data_type = None
+        gnomad = None
+        table_path = args.table_path
+        ht = hl.read_table(table_path)
+   
+    if args.test: 
+        logger.info('Filtering to chr21 for testing')
+        if args.build == 38:
+            ht = ht.filter(ht.locus.contig == '21')
+        else:
+            ht = ht.filter(ht.locus.contig == 'chr21')
+
+    logger.info('Adding chain file and loading fasta sequence for destination build')
+    if args.build == 38:
+
+        rg37.add_liftover('gs://hail-common/references/grch37_to_grch38.over.chain.gz', rg38)
```

Comment:
Not necessary, but I'd probably reframe these as "source" and "target" (getting the right one each at line 101), and then you can avoid duplicating code in the if/else block.